File: SDKs/amazonaws.com/codestar-notifications/2019-10-15/go/pkg/models/operations/listtargets.go
package operations

import (
	"openapi/pkg/models/shared"
)

type ListTargetsQueryParams struct {
	MaxResults *string `queryParam:"style=form,explode=true,name=MaxResults"`
	NextToken  *string `queryParam:"style=form,explode=true,name=NextToken"`
}

type ListTargetsHeaders struct {
	XAmzAlgorithm     *string `header:"name=X-Amz-Algorithm"`
	XAmzContentSha256 *string `header:"name=X-Amz-Content-Sha256"`
	XAmzCredential    *string `header:"name=X-Amz-Credential"`
	XAmzDate          *string `header:"name=X-Amz-Date"`
	XAmzSecurityToken *string `header:"name=X-Amz-Security-Token"`
	XAmzSignature     *string `header:"name=X-Amz-Signature"`
	XAmzSignedHeaders *string `header:"name=X-Amz-SignedHeaders"`
}

type ListTargetsRequestBody struct {
	Filters    []shared.ListTargetsFilter `json:"Filters,omitempty"`
	MaxResults *int64                     `json:"MaxResults,omitempty"`
	NextToken  *string                    `json:"NextToken,omitempty"`
}

type ListTargetsRequest struct {
	QueryParams ListTargetsQueryParams
	Headers     ListTargetsHeaders
	Request     ListTargetsRequestBody `request:"mediaType=application/json"`
}

type ListTargetsResponse struct {
	ContentType               string
	InvalidNextTokenException *interface{}
	ListTargetsResult         *shared.ListTargetsResult
	StatusCode                int64
	ValidationException       *interface{}
}

File: SDKs/amazonaws.com/codestar-notifications/2019-10-15/typescript/src/sdk/models/operations/listtargets.ts
import { Metadata, SpeakeasyBase } from "../../../internal/utils/utils";
import * as shared from "../shared";


export class ListTargetsQueryParams extends SpeakeasyBase {
  @Metadata({ data: "queryParam, style=form;explode=true;name=MaxResults" })
  maxResults?: string;

  @Metadata({ data: "queryParam, style=form;explode=true;name=NextToken" })
  nextToken?: string;
}


export class ListTargetsHeaders extends SpeakeasyBase {
  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Algorithm" })
  xAmzAlgorithm?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Content-Sha256" })
  xAmzContentSha256?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Credential" })
  xAmzCredential?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Date" })
  xAmzDate?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Security-Token" })
  xAmzSecurityToken?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Signature" })
  xAmzSignature?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-SignedHeaders" })
  xAmzSignedHeaders?: string;
}


export class ListTargetsRequestBody extends SpeakeasyBase {
  @Metadata({ data: "json, name=Filters", elemType: shared.ListTargetsFilter })
  filters?: shared.ListTargetsFilter[];

  @Metadata({ data: "json, name=MaxResults" })
  maxResults?: number;

  @Metadata({ data: "json, name=NextToken" })
  nextToken?: string;
}


export class ListTargetsRequest extends SpeakeasyBase {
  @Metadata()
  queryParams: ListTargetsQueryParams;

  @Metadata()
  headers: ListTargetsHeaders;

  @Metadata({ data: "request, media_type=application/json" })
  request: ListTargetsRequestBody;
}


export class ListTargetsResponse extends SpeakeasyBase {
  @Metadata()
  contentType: string;

  @Metadata()
  invalidNextTokenException?: any;

  @Metadata()
  listTargetsResult?: shared.ListTargetsResult;

  @Metadata()
  statusCode: number;

  @Metadata()
  validationException?: any;
}
